testing/helpers/index.js

```diff
@@ -62,3 +62,54 @@ export async function awaitFrameAfter(ms) {
   await sleep(ms);
   await afterRenderPromise();
 }
+
+const VOID_ELEMENTS = new Set([
+  'AREA',
+  'BASE',
+  'BR',
+  'COL',
+  'EMBED',
+  'HR',
+  'IMG',
+  'INPUT',
+  'LINK',
+  'META',
+  'PARAM',
+  'SOURCE',
+  'TRACK',
+  'WBR',
+]);
+
+/**
+ * Returns true of node is a void element.
+ * @param {Node} node
+ * @return {boolean}
+ */
+function isVoidElement(node) {
+  return node.nodeType === Node.ELEMENT_NODE && VOID_ELEMENTS.has(node);
+}
+
+/**
+ * Returns the outerHTML for an element, but with lexicographically sorted attributes.
+ * @param {Node} node
+ * @return {string}
+ */
+export function getDeterministicOuterHTML(node) {
+  const tag = node.localName || node.tagName;
+  const attributes = Array.from(node.attributes)
+    .map(({name, value}) => `${name}="${value}"`)
+    .sort()
+    .join(' ');
+  const start = `<${tag} ${attributes}>`;
+  const contents = Array.from(node.childNodes).map((childNode) => {
+    if (childNode.nodeType === Node.ELEMENT_NODE) {
+      return getDeterministicOuterHTML(childNode);
+    }
+    return childNode.innerHTML || childNode.textContent;
+  });
```

Comment:
```suggestion
  const contents = Array.from(node.childNodes)
    .map((childNode) => childNode.nodeType === Node.ELEMENT_NODE
      ? getDeterministicOuterHTML(childNode)
      : childNode.innerHTML || childNode.textContent);
```

Reply:
I'm not a ternary guy. I don't like how they scale over time (i.e. more code branches means breaking into if/else or god-forbid nested ternaries)

Reply:
Yeah I know, we always differ on this 😆